api/org/commcare/api/persistence/SqlStorageIterator.java

```diff
@@ -0,0 +1,104 @@
+package org.commcare.api.persistence;
+
+import org.javarosa.core.services.storage.IStorageIterator;
+import org.javarosa.core.services.storage.Persistable;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Iterator;
+
+/**
+ * Created by wpride1 on 6/25/15.
+ */
+public class SqlStorageIterator<E extends Persistable> implements IStorageIterator<E>, Iterator<E> {
```

Comment:
I don't like the fact that there is a class with the same name in the odk repository. Can you give them distinct names, or is there a reason they are the same?